atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/IndexDefinition.java

```diff
@@ -117,9 +117,17 @@ public void componentFromDynamicColumn(String componentName, ValueType valueType
      * end of the row
      */
     public void hashFirstRowComponent() {
+        hashFirstNRowComponents(1);
```

Comment:
have the preconditions check here as well, or extract to a method `checkhashRowPreconditions` and call in both methods.